### PR TITLE
[Feature][transforms-v2] Support append only stream from cdc source

### PR DESCRIPTION
**Pull request by hawk9821**
### Purpose of this pull request
issues: #7756 
Add a new transform which can transform cdc row to append only row that contains the cdc rowkind or other needed info.

Example:
CDC row: +D 1, test1, test2
new Row: +I 1,test1,test2,DELETE

### How was this patch tested?
UT:  CdcRowTransformFactoryTest   CdcRowTransformTest
ETE:  TestCdcRowTransformIT

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by cobolbaby** [comment]
Is it possible to use the current SQL transform to turn one record into multiple records?

**Comment by Hisoka-X** [comment]
cc @dailai and @hailin0

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/cdc/CdcRowTransformConfig.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+import java.io.Serializable;
+
+@Getter
+@Setter
+public class CdcRowTransformConfig implements Serializable {
+
+    public static final String PLUGIN_NAME = "CdcRow";
```
Comment (dailai) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/cdc/CdcRowTransformConfig.java]: It's a weird and unintuitive name

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/cdc/CdcRowTransformConfig.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    public static final Option<String> CUSTOM_FIELD_NAME =
+            Options.key("custom_field_name")
+                    .stringType()
+                    .defaultValue("operate_type")
```
Comment (dailai) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/cdc/CdcRowTransformConfig.java]: rowkind is recommended instead

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/cdc/CdcRowTransform.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    protected Column getOutputColumn() {
+        String customFieldName = config.get(CdcRowTransformConfig.CUSTOM_FIELD_NAME);
+        String[] fieldNames = inputCatalogTable.getTableSchema().getFieldNames();
+        boolean isExist = Arrays.asList(fieldNames).contains(customFieldName);
+        if (isExist) {
+            throw new IllegalArgumentException(
+                    String.format("field name %s already exists", customFieldName));
+        }
+        return PhysicalColumn.of(
+                customFieldName, BasicType.STRING_TYPE, (Long) null, true, null, null);
```
Comment (dailai) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/cdc/CdcRowTransform.java]: column length should not be null.
nullable  shoud be false.
default value should not be null.

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/cdc/CdcRowTransformConfig.java (hunk 1 above)
Comment (hawk9821) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/cdc/CdcRowTransformConfig.java]: @Hisoka-X @hailin0  What name would be more appropriate to discuss this converter

Comment (Hisoka-X) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/cdc/CdcRowTransformConfig.java]: how about `RowKindExtractor`

Comment (hawk9821) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/cdc/CdcRowTransformConfig.java]: I am considering extending other CDC fields later, such as ts_ms, database_name, table_name, etc.

Comment (Hisoka-X) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/cdc/CdcRowTransformConfig.java]: > `ts_ms, database_name, table_name`

Not only cdc contains these fields, so I think use name without `CDC` should be better.

**Comment by hawk9821** [comment]
> Is it possible to use the current SQL transform to turn one record into multiple records?
the transformation logic is not consistent with the SQL transform

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/cdc/CdcRowTransformConfig.java (hunk 1 above)
Comment (hawk9821) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/cdc/CdcRowTransformConfig.java]: get.

**Comment by dailai** [comment]
Please fix ci.

File: seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-1/src/test/java/org/apache/seatunnel/e2e/transform/TestRowKindExtractorTransformIT.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import org.apache.seatunnel.e2e.common.container.TestContainer;
+import org.apache.seatunnel.e2e.common.junit.DisabledOnContainer;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.TestTemplate;
+import org.testcontainers.containers.Container;
+
+import java.io.IOException;
+
+@DisabledOnContainer(
```
Comment (dailai) [inline seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-1/src/test/java/org/apache/seatunnel/e2e/transform/TestRowKindExtractorTransformIT.java]: Why diabled spark and flink?

**Review by dailai — DISMISSED** [review]
LGTM If ci passed

**Review by Hisoka-X — DISMISSED** [review]

**Review by dailai — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]
Thanks @hawk9821 and @dailai for review.

**Pull request merged**